docs/introduction/friendly-url.md

```diff
@@ -9,28 +9,39 @@ The rest of the work is done automatically and URLs provided by these providers
-- this interface requires you to implement two methods *(see [ProductDetailFriendlyUrlDataProvider](https://github.com/shopsys/shopsys/blob/9.0/packages/framework/src/Model/Product/ProductDetailFriendlyUrlDataProvider.php) class as an example of the implementation)*:
+- Create new class with name ending with `FriendlyUrlDataProvider` implementing `FriendlyUrlDataProviderInterface` interface
+
+- This interface requires you to implement two methods *(see [ProductDetailFriendlyUrlDataProvider](https://github.com/shopsys/shopsys/blob/9.0/packages/framework/src/Model/Product/ProductDetailFriendlyUrlDataProvider.php) class as an example of the implementation)*:
+
     - `getFriendlyUrlData` method that generates `FriendlyUrlData` for all your entities
+
     - `getRouteName` method returns name of route that you have to declare in `routing_friendly_url.yml` file like:
-        ```
-        front_<entity_name>_detail:
-            path: friendly-url
-            defaults: { _controller: ShopsysShopBundle:Front\<entity_name>:detail }
-        ```
-        this will route all URLs you have provided in `getFriendlyUrlData` method to `<entity_name>Controller::detailAction` method and provide you with ID of entity matching that URL
-        *(see [Symfony Documentation](https://symfony.com/doc/3.4/controller.html) to learn how to create new `Controller`)*
+
+```yaml
+front_<entity_name>_detail:
+    path: friendly-url
+    defaults: { _controller: App\Controller\Front\<entity_name>Controller:detailAction }
+```
+
+this will route all URLs you have provided in `getFriendlyUrlData` method to `<entity_name>Controller::detailAction` method and provide you with ID of entity matching that URL
+*(see [Symfony Documentation](https://symfony.com/doc/3.4/controller.html) to learn how to create new `Controller`)*
+
 - update your `<entity_name>Facade`:
+
     - in `create` and `edit` methods, after calling `$this->em->flush()`, add similar code like this:
-        ```
-        $this->friendlyUrlFacade->createFriendlyUrls('front_<entity_name>_detail', $entity->getId(), $entity->getNames());
-        ```
-        - third parameter is an array of names indexed by locale that will be used for URL generation (e.g. 'cs' => 'Televize', 'en' => 'Television')
-    - this way you will have always generated friendly URLs based on your entity name
-        - if the name of the entity will be changed, the old URLs will now redirect to your new URLs
+
+```php
+// third parameter is an array of names indexed by locale that will be used for URL generation (e.g. 'cs' => 'Televize', 'en' => 'Television')
+$this->friendlyUrlFacade->createFriendlyUrls('front_<entity_name>_detail', $entity->getId(), $entity->getNames());
+
+```
+
+this way you will have always generated friendly URLs based on your entity name and if the name of the entity will be changed, the old URLs will now redirect to your new URLs
+
 - run `php phing friendly-urls-generate` to generate new URLs
+
```

Comment:
It's because code-block inside a list is not rendered properly. 
See the current version https://docs.shopsys.com/en/9.0/introduction/friendly-url/

it's even mentioned in guidelines for writing documentation https://docs.shopsys.com/en/9.0/contributing/guidelines-for-writing-documentation/

>Rendering of multi-line code-blocks in the lists does not work properly so if you need to add such thing in your article, consider using headlines instead of bullet points in your text flow, or just do not indent the code-blocks at all in your list.

Reply:
It is ok for me